Code/Framework/AzCore/AzCore/IO/TextStreamWriters.h

```diff
@@ -5,184 +5,193 @@
-#define AZCORE_IO_TEXTSTREAMWRITERS_H
+#pragma once
 
 #include <AzCore/base.h>
 #include <AzCore/IO/GenericStreams.h>
 #include <AzCore/std/containers/vector.h>
 
-namespace AZ
+namespace AZ::IO
 {
-    namespace IO
+    /**
+     * For use with rapidxml::print.
+     */
+    class RapidXMLStreamWriter
     {
-        /**
-         * For use with rapidxml::print.
-         */
-        class RapidXMLStreamWriter
+    public:
+        class PrintIterator
         {
         public:
-            class PrintIterator
+            PrintIterator(RapidXMLStreamWriter* writer)
+                : m_writer(writer)
             {
-            public:
-                PrintIterator(RapidXMLStreamWriter* writer)
-                    : m_writer(writer)
-                {
-                }
+            }
 
-                PrintIterator(const PrintIterator& it)
-                    : m_writer(it.m_writer)
-                {
-                }
+            PrintIterator(const PrintIterator& it)
+                : m_writer(it.m_writer)
+            {
+            }
 
-                PrintIterator& operator=(const PrintIterator& rhs) = default;
+            PrintIterator& operator=(const PrintIterator& rhs) = default;
 
-                PrintIterator& operator*() { return *this; }
-                PrintIterator& operator++() { return *this; }
-                PrintIterator& operator++(int) { return *this; }
-                PrintIterator& operator=(char c)
+            PrintIterator& operator*() { return *this; }
+            PrintIterator& operator++() { return *this; }
+            PrintIterator& operator++(int) { return *this; }
+            PrintIterator& operator=(char c)
+            {
+                if (m_writer->m_cache.size() == m_writer->m_cache.capacity())
                 {
-                    if (m_writer->m_cache.size() == m_writer->m_cache.capacity())
-                    {
-                        m_writer->FlushCache();
-                    }
-                    m_writer->m_cache.push_back(c);
-                    return *this;
+                    m_writer->FlushCache();
                 }
-            private:
-                RapidXMLStreamWriter* m_writer;
-            };
-
-            RapidXMLStreamWriter(IO::GenericStream* stream, size_t writeCacheSize = 128 * 1024)
-                : m_stream(stream)
-                , m_errorCount(0) 
-            {
-                m_cache.reserve(writeCacheSize);
+                m_writer->m_cache.push_back(c);
+                return *this;
             }
+        private:
+            RapidXMLStreamWriter* m_writer;
+        };
 
-            ~RapidXMLStreamWriter()
-            {
-                FlushCache();
-            }
+        RapidXMLStreamWriter(IO::GenericStream* stream, size_t writeCacheSize = 128 * 1024)
+            : m_stream(stream)
+            , m_errorCount(0)
+        {
+            m_cache.reserve(writeCacheSize);
+        }
+
+        ~RapidXMLStreamWriter()
```

Comment:
nit:
If we are not declaring this as a virtual desctrutor, then maybe mark this class ```final```?